yoyodyne/defaults.py

```diff
@@ -35,6 +35,12 @@
 MAX_TARGET_LENGTH = 128
 EVAL_METRICS = set()
 
+# Tuning arguments.
+AUTO_BATCH_SIZE_FIND = False
+AUTO_BATCH_SIZE_MODE = "power"
+AUTO_BATCH_SIZE_STEPS_PER_TRIAL = 3
```

Comment:
Is there justification for 3 rather than 1? Furthermore for any number < number of steps per epoch, we probably can't ensure you won't run OOM right?

Reply:
This is just shadowing the default values from the lower-level PTL library. 

The `"power"` method then just starts with 2^1, 2^2, etc. until it hits an OOM or batch size exceeds the training set size. If you use `"binsearch"` it starts like that using the OOM size as an upper bound. The extra work in the binary search method is usually not worth it if you're going to use gradient accumulation since the extra work it does will move it away from a nice power-of-two value with good divisors. 

Yes, this is heuristic when less than the number of batches per epoch. If you get an OOM anyways, try doubling it. (I actually hit that case once; doubling fixed it.)

Reply:
Ahhh ok makes sense!